04-Games/src/js/games/FallingStones.js

```diff
@@ -38,18 +39,21 @@ export class FallingStones extends GameTemplate {
     update(ctx) {
         this.player.update(ctx);
         this.player.borderCollision(ctx);
-        for (let i = 0; i < this.bullets.length; ++i) {
+        for (let i = this.bullets.length -1 ; i >= 0 ; i--) {
             this.bullets[i].update(ctx); //update coordinates of all projectiles
-            for (let m = 0; m < this.stones.length; ++m) {
+            for (let m = this.stones.length -1; m >= 0; m--) {
```

Comment:
weiter unten hast du nochmal eine for schleife die über stones iteriert, halbier doch deinen rechenaufwand und fass sie zusammen.